actionview/lib/action_view/helpers/form_with_helper.rb

```diff
@@ -0,0 +1,196 @@
+module ActionView
+  module Helpers
+    module FormWithHelper
+      class FormWithBuilder
+        include ActionView::Helpers::TagHelper
+
+        FIELD_HELPERS = [:fields_for, :label, :check_box, :radio_button,
+                        :hidden_field, :file_field, :text_field,
+                        :password_field, :color_field, :search_field,
+                        :telephone_field, :phone_field, :date_field,
+                        :time_field, :datetime_field, :datetime_local_field,
+                        :month_field, :week_field, :url_field, :email_field,
+                        :number_field, :range_field]
+
+        GENERATED_FIELD_HELPERS = FIELD_HELPERS - [:label, :check_box, :radio_button, :fields_for, :file_field, :text_area]
+
+        attr_accessor :model, :scope, :url, :remote
+
+        GENERATED_FIELD_HELPERS.each do |selector|
+          class_eval <<-RUBY_EVAL, __FILE__, __LINE__ + 1
+            def #{selector}(method, *args, **options)                             # def text_field(method, *args, **options)
+              tag.input options_for_field('#{selector}', method, args, options)   #   tag.input options_for_field(selector, method, args, options)
+            end                                                                   # end
+          RUBY_EVAL
+        end
+
+        def file_field(method, *args, **options)
+          multiple = options.slice(:multiple)
+          options = options_for(method, options).merge!(type: "file")
+          options.merge!(multiple)
+          tag.input options
+        end
+
+        def submit
+          tag.input value: submit_default_value, "data-disable-with": submit_default_value, type: 'submit', name: 'commit'
+        end
+
+        def text_area(method, *args, **options)
+          options = options_for(method, options)
+          content = model.send(method) if model
+          content = args[0] if args.size > 0
+          content.nil? ? tag.textarea(options) : tag.textarea(content, options)
+        end
+
+        def label(method, *args, **options, &block)
+          content ||= args[0] if args.size > 0
+          content ||= I18n.t("#{model.model_name.i18n_key}.#{method}", default: "", scope: "helpers.label").presence if model && model.model_name
+          content ||= translate_with_human_attribute_name(method)
+          content ||= method.to_s.humanize
+          content = @template.capture(content, &block) if block_given?
+          tag.label content, options
+        end
+
+        def check_box(method, include_hidden: true, on: "1", off: "0", **options)
```

Comment:
`method` confuses me because I keep thinking of the `_method` input field as well as the HTTP methods.

To me it makes much more sense to say `attribute` since that what we're working on.